tests/test_core.py

```diff
@@ -128,6 +128,18 @@ def test_init_df_from_timeseries(test_df):
     pd.testing.assert_frame_equal(df.timeseries(), test_df.timeseries())
 
 
+def test_init_df_from_timeseries_unused_levels(test_df):
+    # this test guards against regression for the bug
+    # reported in https://github.com/IAMconsortium/pyam/issues/762
+
+    for _, df in test_df.timeseries().groupby(["model", "scenario"]):
+        # we're only interested in the second model-scenario combination
+        pass
+
+    # pandas does not remove unused levels (here: "Primary Energy|Coal") in a groupby
+    assert IamDataFrame(df).variable == ["Primary Energy"]
```

Comment:
```suggestion
    for (model, scenario), df in test_df.timeseries().groupby(["model", "scenario"]):
        # we're only interested in this particular model-scenario combination
        # because it doesn't have Primary Energy|Coal
        if (model == "model_a" and scenario == "scen_b"):
            # pandas does not remove unused levels (here: "Primary Energy|Coal") in a groupby
            assert IamDataFrame(df).variable == ["Primary Energy"]
```

I don't know if groupby guarantees order so I always get nervous assuming it does. I think this change is a bit more defensive but also fine to leave how it is if I'm jumping at shadows.

Reply:
Thanks! Implemented your suggestion slightly differently to ensure that the test is actually triggered - your approach might end up never calling the assertion because it's inside the if-statement...

Reply:
Smart